processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -133,13 +168,16 @@ protected AddToFactsResult addToFacts(
 
       // Last ditch sanity checks
       if (numEntries.get() >= maxRowCount
-          && facts.getPriorIndex(key) == TimeAndDims.EMPTY_ROW_INDEX
-          && !skipMaxRowsInMemoryCheck) {
-        throw new IndexSizeExceededException("Maximum number of rows [%d] reached", maxRowCount);
+          || maxBytesInMemory != -1 && sizeInBytes.get() >= maxBytesInMemory
+             && facts.getPriorIndex(key) == TimeAndDims.EMPTY_ROW_INDEX
+             && !skipMaxRowsInMemoryCheck) {
```

Comment:
Looks like there are a couple of newlines missing here- might need a run through the code styler.